### PR TITLE
Log a warning when an executable is found, but does not have the exec…

### PR DESCRIPTION
**Pull request by pmrv**
… bit set

### CODE REVIEW

**Comment by pmrv** [comment]
Fixes #21.

File: tests/unit/test_resources.py [hunk 1]
```diff
@@ -56,6 +57,8 @@ def test_executable(self):
             with self.subTest(suffix=suffix):
                 res = ExecutableResolver([self.res1], code="code1", module="module1", suffix=suffix)
                 if os.name != "nt":
+                    with self.assertLogs(logger, level="WARNING"):
+                        res.list()
```
[the remark this answers is not in the document]
Comment (pmrv) [reply tests/unit/test_resources.py:61]: Reading around I understood that windows always reports the exec bit set, but I don't have python on the one windows machine I have.

[the remark this answers is not in the document]
Comment (jan-janssen) [reply tests/unit/test_resources.py:61]: I had the same feeling. Can you add a corresponding comment in the code just to make sure we remember it the next time?

**Comment by pmrv** [comment]
> I'm uncomfortable with the cross dependency to another part of the snippets package-- even if it is locked away and not top level.
> 
> I need to check, but if we import the logger and then just warnings.warn (or maybe (re)import the default python logger and pass a warning there) doesn't this get caught by the pyiron log? I thought it should bz virtue of the logger module configuring the base logger, but maybe I'm being naive about a fresh instance of the logger ignoring this

Fair, I think this should be possible, but will need to check.

**Comment by liamhuber** [comment]
> Fair, I think this should be possible, but will need to check.

If it doesn't work pretty straightforwardly, I am content to make an exception (especially since it's not a top-level import), but let's check if we can have our cake and eat it too.

**Comment by pmrv** [comment]
The standard library has [this](https://github.com/python/cpython/blob/8d1f15712d243e03e50332abb86ff4c121814aa5/Lib/logging/__init__.py#L2331), with the caveat that it redirects all warnings and to a hard coded logger, but it would be possible to setup similar logic in snippets.logger and redirect all warnings or warnings of a certain type to the pyiron logger.  The warning type would have to be defined in a shared module somewhere so still link logger and resources.

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by liamhuber** [comment]
> The standard library has [this](https://github.com/python/cpython/blob/8d1f15712d243e03e50332abb86ff4c121814aa5/Lib/logging/__init__.py#L2331), with the caveat that it redirects all warnings and to a hard coded logger, but it would be possible to setup similar logic in snippets.logger and redirect all warnings or warnings of a certain type to the pyiron logger. The warning type would have to be defined in a shared module somewhere so still link logger and resources.

That's unfortunate. But now playing with it myself and thinking about it more, I really, really don't like invoking the snippets logger here. Anyone hitting this warning will suddenly get a `pyiron.log` file on their filesystem. This seems uncool for people importing a random (not-`logger`) class from `pyiron_snippets`.

An alternative would be something like sticking a `._logger = None` attribute on the relevant class and then internally doing an `if self._logger is None: warnings.warn(msg); else: self._logger.warning(msg)`, and manually setting the logger attribute when we're using the class where the pyiron logger is available? I feel this is still ugly, but less evil than writing surprise files for people.

**Comment by pmrv** [comment]
Agree about the uncleanliness when not using the logger otherwise, so I reverted back to warning.  I've added a custom warning type, so that we could potential filter and redirect it to the logger in the future, but I'm not too married to it.

**Review by liamhuber — APPROVED** [review]

**Pull request merged**